osquery/tables/applications/linux/docker_deb_packages.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/applications/linux/namespace_ops.h>
+#include <osquery/tables/applications/posix/docker_api.h>
+#include <osquery/tables/system/linux/deb.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include <iostream>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include "osquery/core/json.h"
+#include "rapidjson/filereadstream.h"
+
+namespace osquery {
+namespace tables {
+
+void extractDebPackageInfo(const struct pkginfo* pkg,
+                           rapidjson::Writer<rapidjson::StringBuffer>& w) {
+  struct varbuf vb;
+  varbuf_init(&vb, 20);
+  w.StartObject();
+
+  // Iterate over the desired fieldinfos, calling their fwritefunctions
+  // to extract the package's information.
+  const struct fieldinfo* fip = nullptr;
+  for (fip = fieldinfos; fip->name; fip++) {
```

Comment:
If fieldinfos were an STL collection, it'd be quite easy to use range-based for syntax here, like:
```
for (const fieldinfo& fi : fieldinfos) {
    fi.wcall(&vb, pkg, &pkg->installed, ... etc ...
    ... etc ...
}
```